frontends/web/src/routes/account/add/add.tsx

```diff
@@ -57,81 +61,99 @@ class ManageAccount extends Component<Props, State> {
-                case 1: return ({ step: 0 });
-                case 2: return ({ step: 1 });
-            }
-        });
+        switch (this.state.step) {
+            case 'choose-name':
+                this.setState({ step: 'select-coin' });
+                break;
+            case 'success':
+                this.setState({ step: 'choose-name' });
+                break;
+        }
     }
 
-    private next = () => {
-        const { accountName, coinCode } = this.state;
-        switch (this.getStep()) {
+    private next = (e: Event) => {
+        e.preventDefault();
+        const { accountCode, accountName, coinCode, step } = this.state;
+        const { t } = this.props;
+        switch (step) {
             case 'select-coin':
-                console.info(`${coinCode} selected`);
-                this.setState({ step: 1 });
+                this.setState({ step: 'choose-name' });
                 break;
             case 'choose-name':
                 interface ResponseData {
                     success: boolean;
+                    accountCode?: string;
                     errorCode?: 'alreadyExists' | 'limitReached';
                     errorMessage?: string;
                 }
 
                 apiPost('account-add', {
                     coinCode,
                     name: accountName,
-                }).then((data: ResponseData) => {
-                    if (data.success) {
-                        this.setState({ step: 2 });
-                        //route('/account/' + data.accountCode);
-                    } else {
-                        if (data.errorCode) {
-                            alertUser(this.props.t(`addAccount.error.${data.errorCode}`));
+                })
+                    .then((data: ResponseData) => {
+                        if (data.success) {
+                            this.setState({
+                                accountCode: data.accountCode,
+                                step: 'success'
+                            });
+                            //route('/account/' + data.accountCode);
+                        } else if (data.errorCode) {
+                            this.setState({
+                                errorMessage: t(`addAccount.error.${data.errorCode}`)
+                            });
                         } else if (data.errorMessage) {
-                            alertUser(this.props.t('unknownError', { errorMessage: data.errorMessage }));
+                            this.setState({
+                                errorMessage: t('unknownError', { errorMessage: data.errorMessage })
+                            });
                         }
-                    }
-                });
+                    });
                 break;
             case 'success':
+                if (accountCode) {
+                    route(`/account/${accountCode}`);
+                    return;
+                }
                 route('/account-summary');
```

Comment:
i can't see when this happens - seems success=true is set at the same time as accountCode, always?

Reply:
From your other PR I assumed accountCode isn't always there https://github.com/digitalbitbox/bitbox-wallet-app/pull/1279/files#diff-74c54a5fb0b456d69855bb21b08f65442ab1db6a07ee52185345853d4eb641c0R116

Shall I remove `if (accountCode) {` and expect backend to always return accountCode?